### ISSUE TITLE
Critical Bug: Quantization Fails on Custom LSTM Models in Model Optimization Library

### ISSUE THREAD

**Issue by ShengDong207** [tensorflow/tensorflow#94100]
## Bug Description
Quantization is failing when applied to custom LSTM models using the TensorFlow model optimization library.

## Environment
- TensorFlow version: [Please specify your TF version]
- Model Optimization Library version: [Please specify your version]
- Operating System: [Please specify your OS]
- Python version: [Please specify your Python version]

## Steps to Reproduce
1. Define a custom LSTM model with the following architecture:
   ```python
   def create_custom_lstm_model(input_shape, units=64):
       inputs = tf.keras.Input(shape=input_shape)
       x = tf.keras.layers.LSTM(units, return_sequences=True)(inputs)
       x = tf.keras.layers.LSTM(units)(x)
       outputs = tf.keras.layers.Dense(1)(x)
       return tf.keras.Model(inputs=inputs, outputs=outputs)
   ```

2. Train the model on sample data:
   ```python
   model = create_custom_lstm_model((sequence_length, features))
   model.compile(optimizer='adam', loss='mse')
   model.fit(x_train, y_train, epochs=5)
   ```

3. Attempt to quantize the model using the TF Model Optimization toolkit:
   ```python
   import tensorflow_model_optimization as tfmot
   
   quantize_model = tfmot.quantization.keras.quantize_model
   
   q_aware_model = quantize_model(model)
   q_aware_model.compile(optimizer='adam', loss='mse')
   q_aware_model.fit(x_train, y_train, epochs=3)  # Fine-tune with quantization awareness
   
   converter = tf.lite.TFLiteConverter.from_keras_model(q_aware_model)
   converter.optimizations = [tf.lite.Optimize.DEFAULT]
   quantized_tflite_model = converter.convert()
   ```

## Expected Behavior
The quantization process should complete successfully, generating a quantized TFLite model with reduced size that maintains reasonable accuracy.

## Actual Behavior
The quantization process fails with the following error:
[Please provide the exact error message you're encountering]

## Additional Information
- The same quantization process works correctly on simpler models without LSTM layers
- The issue appears to be specific to custom LSTM implementations rather than the built-in LSTM layers
- The error occurs consistently across multiple attempts and model configurations

## Potential Impact
This bug blocks the deployment of optimized LSTM models in resource-constrained environments, which is a critical use case for the model optimization library.

Please provide any additional information that might help reproduce and diagnose this issue.

**Comment by Venkat6871** [comment]
Please stop spamming.
